File: app/src/main/java/com/soho/sohoapp/feature/marketplaceview/feature/filterview/PropertyFilterSavedAdapter.java
package com.soho.sohoapp.feature.marketplaceview.feature.filterview;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.soho.sohoapp.BaseViewHolder;
import com.soho.sohoapp.R;
import com.soho.sohoapp.feature.home.BaseModel;

import java.util.List;

import butterknife.BindView;

/**
 * Created by chowii on 25/8/17.
 */

class PropertyFilterSavedAdapter extends RecyclerView.Adapter<BaseViewHolder> {

    private final List<String> jsonObject;

    public PropertyFilterSavedAdapter(List<String> jsonObject) { this.jsonObject = jsonObject; }

    @Override
    public int getItemCount() { return jsonObject.size() + 1; }

    @Override
    public int getItemViewType(int position) {
        if(position == 0) return R.layout.item_filter_header;
        else return R.layout.item_filter_saved;
    }

    @Override
    public BaseViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View itemView = LayoutInflater.from(parent.getContext()).inflate(viewType, parent, false);
        switch(viewType){
            case R.layout.item_filter_header:
                return new FilterSavedHeaderViewHolder(itemView);
            case R.layout.item_filter_saved:
                return new SavedItemViewHolder(itemView);
        }
        return null;
    }

    @Override
    public void onBindViewHolder(BaseViewHolder holder, int position) {
        if(holder instanceof SavedItemViewHolder){
            SavedItemViewHolder item = (SavedItemViewHolder) holder;
            item.titleTextView.setText("Filter " + position);
            item.subtitleTextView.setText(jsonObject.get(position-1));
        }
    }

    class SavedItemViewHolder extends BaseViewHolder{

        @BindView(R.id.title_text_view)
        TextView titleTextView;

        @BindView(R.id.subtitle_text_view)
        TextView subtitleTextView;

        SavedItemViewHolder(View itemView) { super(itemView); }

        @Override
        public void onBindViewHolder(BaseModel model) { }
    }
}

File: app/src/main/java/com/soho/sohoapp/feature/marketplaceview/feature/filters/savedfilters/PropertyFilterSavedAdapter.kt
package com.soho.sohoapp.feature.marketplaceview.feature.filters.savedfilters

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import butterknife.BindView
import butterknife.ButterKnife
import com.soho.sohoapp.R
import com.soho.sohoapp.database.entities.MarketplaceFilterWithSuburbs

/**
 * Created by chowii on 25/8/17.
 */
class PropertyFilterSavedAdapter(val onClickListener : PropertyFilterOnClickListener) :
        RecyclerView.Adapter<PropertyFilterSavedAdapter.SavedItemViewHolder>() {

    private lateinit var filters : List<MarketplaceFilterWithSuburbs>

    // MARK: - ================== RecyclerView.Adapter related ==================

    override fun getItemCount(): Int = filters.size

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) =
            SavedItemViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.item_filter_saved, parent, false))

    override fun onBindViewHolder(holder: SavedItemViewHolder?, position: Int) {
        holder?.bind(filters[position])
        holder?.itemView?.setOnClickListener {
            onClickListener.filterSelected(filters[position])
        }
    }

    // MARK: - ================== Additional methods ==================

    fun updateDataSet(newDataSet:List<MarketplaceFilterWithSuburbs>) {
        this.filters = newDataSet
        notifyDataSetChanged()
    }

    fun getFilterAt(index:Int) : MarketplaceFilterWithSuburbs =
            filters[index]

    // MARK: - ================== ViewHolder ==================

    class SavedItemViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        @BindView(R.id.subtitle_text_view)
        lateinit var subtitleTextView : TextView

        @BindView(R.id.title_text_view)
        lateinit var titleTextView : TextView

        init {
            ButterKnife.bind(this, itemView)
        }

        fun bind(filter:MarketplaceFilterWithSuburbs) {
            titleTextView.text = filter.suburbsDisplayString(
                    itemView.context.getString(R.string.filters_multiple_suburbs_format),
                    itemView.context.getString(R.string.filters_all_suburbs))
            subtitleTextView.text = filter.marketplaceFilter.priceRangeDisplayString(
                    itemView.context.getString(R.string.filters_search_bar_display_format),
                    itemView.context.getString(R.string.dollar_format),
                    itemView.context.getString(R.string.filters_price_any))
        }
    }

    interface PropertyFilterOnClickListener {
        fun filterSelected(filter:MarketplaceFilterWithSuburbs)
    }
}
